common/component/postgresql/v1/metadata_test.go

```diff
@@ -29,7 +30,8 @@ func TestMetadata(t *testing.T) {
 		m := pgMetadata{}
 		props := map[string]string{}
 
-		err := m.InitWithMetadata(state.Metadata{Base: metadata.Base{Properties: props}}, false)
+		opts := postgresql.InitWithMetadataOpts{AzureADEnabled: false, AWSIAMEnabled: false}
```

Comment:
Nit: no need to specify the default values.

```suggestion
		opts := postgresql.InitWithMetadataOpts{}
```